### ISSUE TITLE
Implement gradient support for complex types

### ISSUE THREAD

**Issue by educhesne** [pymc-devs/pytensor#1366]
### Description

The question arose in PR #1303 to support complex gradient.

There are two conventions regarding complex gradient:
* the holomorphic compatible gradient, as in [JAX](https://docs.jax.dev/en/latest/notebooks/autodiff_cookbook.html#complex-numbers-and-differentiation)
* its conjugate, the gradient-descent compatible gradient, as in [torch](https://pytorch.org/docs/stable/notes/autograd.html#complex-autograd-doc)

At the moment `verify_grad` is not defined on complex gradient, and anyway not all Op accepts complex inputs.

So the current issue can serve two purposes:
* register interests in having complex gradients (and complex Op)
* deciding which complex gradient convention to follow

**Comment by educhesne** [comment]
I've open a draft PR #1367 proposing an implementation of `verify_grad` following the JAX-like convention.
This is just to support the discussion; switching convention is simply removing the conjugate [here](https://github.com/pymc-devs/pytensor/pull/1367/files#diff-d94d878e6b0b8d4ea9005f70f94b7bf5dc6ea26b724cb1231d621ab84dcc46a6R1717)
